backend/api/Services/EchoService.cs

```diff
@@ -0,0 +1,182 @@
+﻿using System.Net.Http.Headers;
+using System.Text.Json;
+using System.Web;
+using Api.Models;
+using Azure.Core;
+using Azure.Identity;
+
+namespace Api.Services
+{
+    public class EchoService
+    {
+        private readonly ILogger<EchoService> _logger;
+
+        private readonly DefaultAzureCredential _credentials;
+
+        private readonly IConfiguration _config;
+
+        private static readonly HttpClient client = new();
+
+        private readonly string _echoApiUrl;
+
+        private readonly string[] _requestScope;
+
+        private readonly string _installationCode;
+
+        public EchoService(ILogger<EchoService> logger, DefaultAzureCredential credentials, IConfiguration config)
+        {
+            _logger = logger;
+            _credentials = credentials;
+            _config = config;
+
+            _echoApiUrl = _config.GetSection("Echo").GetValue<string>("EchoApiUrl");
+            string echoAppScope = _config.GetSection("Echo").GetValue<string>("EchoAppScope");
+            string echoClientId = _config.GetSection("Echo").GetValue<string>("EchoClientId");
+            _installationCode = config.GetValue<string>("InstallationCode");
+
+            _requestScope = new string[1] { $"{echoClientId}/{echoAppScope}" };
+        }
+
+        public async Task<IList<Mission>> GetMissions()
+        {
+            var accessToken = await AcquireAccessToken();
+            ConfigureRequest(accessToken);
+            var uri = ConstructGetMissionsRequestUri();
+
+            var response = await client.GetAsync(uri);
+            if (!response.IsSuccessStatusCode) throw new MissionNotFoundException($"Failed to retrieve missions from Echo: {response.ReasonPhrase}");
```

Comment:
This is a very long line, is it better to split it? I like short if-oneliners, but im not sure if i like the long ones.